test/libsolidity/syntaxTests/parsing/errorRecovery/e7.sol

```diff
@@ -0,0 +1,8 @@
+pragma solidity >=0.0.0;
+
+contract E7 {
+	constructor() public {
+	    balances[tx.origin] = // missing RHS and semicolon.
+	}
+
+}
```

Comment:
No error here?

Reply:
The error here is that there is no RHS or semicolon. Since there is no semicolon to the end of the file, there is no token for `Statement` to synchronize on.

In the current recovery scheme, the code for `Statement` resets the scanning position back to the point of error, `}` and then raises a fatalParserError then which is handled by `Block`. 

Does this make sense?

Reply:
I was wondering why this test file did not require an error. Then I noticed that the pull request does not pass the tests, so it's fine.